src/Domain/Donation.php

```diff
@@ -1321,4 +1316,56 @@ public function cancel(): void
 
         $this->donationStatus = DonationStatus::Cancelled;
     }
+
+    /**
+     * Updates a donation to set the appropriate fees. If card details are null then we assume for now that a card with
+     * the lowest possible fees will be used, and this should be called again with the details of the selected card
+     * when confirming the payment.
+     *
+     * @param string|null $cardBrand
+     * @param string|null $cardCountry ISO two letter uppercase code
+     */
+    public function deriveFees(?string $cardBrand, ?string $cardCountry): void
+    {
+        $incursGiftAidFee = $this->hasGiftAid() && $this->hasTbgShouldProcessGiftAid();
+
+        $fees = Calculator::calculate(
+            $this->getPsp(),
+            $cardBrand,
+            $cardCountry,
+            $this->getAmount(),
+            $this->getCurrencyCode(),
+            $incursGiftAidFee,
+            $this->getCampaign()->getFeePercentage(),
+        );
+
+        $this->setCharityFee($fees->coreFee);
+        $this->setCharityFeeVat($fees->feeVat);
+    }
+
+    public function collectFromStripeCharge(
+        string $chargeId,
+        string $transferId,
+        ?string $cardBrand,
+        ?string $cardCountry,
+        string $originalFeeFractional,
+        int $chargeCreationTimestamp
+    ): void {
+        Assertion::eq(is_null($cardBrand), is_null($cardCountry));
+        Assertion::numeric($originalFeeFractional);
+        Assertion::notEmpty($chargeId);
+        Assertion::notEmpty($transferId);
+
+        $this->chargeId = $chargeId;
+        $this->transferId = $transferId;
+
+        if ($cardBrand) {
+            /** @psalm-var value-of<Calculator::STRIPE_CARD_BRANDS> $cardBrand */
+            $this->deriveFees($cardBrand, $cardCountry);
```

Comment:
this new call to `deriveFees` is the nugget of the PR.  This function `collectFromStripeCharge` is called from the `\MatchBot\Application\Actions\Hooks\StripePaymentsUpdate` handler referenced in the PR title.